fission/logdb/influxdb.go

```diff
@@ -17,47 +17,35 @@ limitations under the License.
 package logdb
 
 import (
-	"log"
-	"time"
-
+	"encoding/json"
 	"fmt"
-
+	"log"
+	"net/http"
+	"net/url"
 	"strings"
+	"time"
 
 	influxdbClient "github.com/influxdata/influxdb/client/v2"
+
+	"github.com/fission/fission"
 )
 
 const (
 	INFLUXDB_DATABASE = "fissionFunctionLog"
+	INFLUXDB_URL      = "http://influxdb:8086/query"
 )
 
-func NewInfluxDB(cnf DBConfig) (InfluxDB, error) {
-	dbClient, err := influxdbClient.NewHTTPClient(influxdbClient.HTTPConfig{
-		Addr:     cnf.Endpoint,
-		Username: cnf.Username,
-		Password: cnf.Password,
-	})
-	if err != nil {
-		log.Fatal(err)
-		return InfluxDB{}, err
-	}
-
-	return InfluxDB{
-		dbClient: dbClient,
-	}, nil
+func NewInfluxDB(serverURL string) (InfluxDB, error) {
+	return InfluxDB{endpoint: serverURL}, nil
 }
 
 type InfluxDB struct {
-	dbClient influxdbClient.Client
+	endpoint string
 }
 
 func (influx InfluxDB) GetPods(filter LogFilter) ([]string, error) {
 	query := fmt.Sprintf("select * from \"log\" where \"funcuid\" = '%s' group by \"pod\"", filter.FuncUid)
```

Comment:
Can we remove this Sprintf, which is vulnerable to "sql" injection, and instead use `influxdb.NewQueryWithParameters`?